packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1102,39 +1102,37 @@ abstract class ResidentRunner {
       'flutter',
       'png',
     );
-    final List<FlutterView> views = await device
-      .vmService.getFlutterViews();
+    List<FlutterView> views = <FlutterView>[];
+    Future<bool> toggleDebugBanner() async {
+      try {
+        for (final FlutterView view in views) {
+          await device.vmService.flutterDebugAllowBanner(
+            false,
```

Comment:
`toggleDebugBanner` should maybe be `setDebugBanner` and accept a bool arg to turn it back on, too.